ndcube/utils/wcs.py

```diff
@@ -361,3 +363,72 @@ def validate_physical_types(physical_types):
             "It must be a string specified in the list (http://www.ivoa.net/documents/latest/UCDlist.html) "
             "or if no matching type exists it can be any string prepended with 'custom:'."
         )
+
+
+def calculate_world_indices_from_axes(wcs, axes):
+    """
+    Given a string representation of a world axis or a numerical array index, convert it
+    to a numerical world index aligning to the position in
+    wcs.world_axis_object_components.
+    """
+    # Convert input axes to WCS world axis indices.
+    world_indices = []
+    for axis in axes:
+        if isinstance(axis, numbers.Integral):
+            # If axis is int, it is a numpy order array axis.
+            # Convert to pixel axis in WCS order.
+            axis = convert_between_array_and_pixel_axes(np.array([axis]), wcs.pixel_n_dim)[0]
+            # Get WCS world axis indices that correspond to the WCS pixel axis
+            # and add to list of indices of WCS world axes whose coords will be returned.
+            world_indices += list(pixel_axis_to_world_axes(axis, wcs.axis_correlation_matrix))
+        elif isinstance(axis, str):
+            # If axis is str, it is a physical type or substring of a physical type.
+            world_indices.append(physical_type_to_world_axis(axis, wcs.world_axis_physical_types))
+        else:
+            raise TypeError(f"Unrecognized axis type: {axis, type(axis)}. "
+                            "Must be of type (numbers.Integral, str)")
+    # Use inferred world axes to extract the desired coord value
+    # and corresponding physical types.
+    return np.unique(np.array(world_indices, dtype=int))
+
+
+def array_indicies_for_world_objects(wcs, axes=None):
```

Comment:
This needs tests

Reply:
Typo in function name.  Should be `indices`